### PR TITLE
EVAKA-4171 bulletin sender dropdown

### PR DESCRIPTION
**Pull request by hruoho**
#### Summary

Let user choose sender from a dropdown instead of using free text input field. Under the hood, this `sender` is just a string and not a reference.

### CODE REVIEW

File: service/src/integrationTest/kotlin/fi/espoo/evaka/messaging/BulletinIntegrationTest.kt [hunk 1]
```diff
@@ -348,6 +348,21 @@ class BulletinIntegrationTest : FullApplicationTest() {
...
+            .responseObject<List<String>>()
+
+        assertEquals(200, res.statusCode)
+
+        val senderOptions = result.get()
+
+        assertEquals(3, senderOptions.size)
+
+        assertEquals(setOf("Elina Esimies", "Testaajat", "Test Daycare"), senderOptions.toSet())
+    }
```
Comment (mikkopiu) [inline service/src/integrationTest/kotlin/fi/espoo/evaka/messaging/BulletinIntegrationTest.kt:364]: I'd also like to see the test repeated for the second unit (that it gets the correct unit and still the same user) -- just as a regression/sanity test.

File: frontend/src/e2e-test/specs/7_messaging/sending-and-receiving-bulletins.spec.ts [hunk 2]
```diff
@@ -116,8 +116,10 @@ test('Supervisor sends bulletin and guardian reads it', async (t) => {
 
   await t.expect(citizenMessages.messageReaderTitle.textContent).eql('Hello')
   await t
-    .expect(citizenMessages.messageReaderSender.textContent)
-    .eql('Test sender')
+    .expect(
+      (await citizenMessages.messageReaderSender.textContent).toLowerCase()
+    )
+    .contains('päiväkoti')
```
Comment (mikkopiu) [inline frontend/src/e2e-test/specs/7_messaging/sending-and-receiving-bulletins.spec.ts:122]: Is a full `eql` check not possible/reasonable here?

File: frontend/src/e2e-test/pages/employee/messaging/messages-page.ts [hunk 3]
```diff
@@ -9,11 +9,8 @@ export default class MessagesPage {
...
 
-  async createNewBulletin(sender: string, title: string, content: string) {
+  async createNewBulletin(title: string, content: string) {
     await t.click(Selector('[data-qa="new-bulletin-btn"]'))
-    await t.typeText(Selector('[data-qa="input-sender"]'), sender, {
-      replace: true
-    })
     await t.typeText(Selector('[data-qa="input-title"]'), title)
     await t.typeText(Selector('[data-qa="input-content"]'), content)
     await t.click(Selector('[data-qa="send-bulletin-btn"]'))
```
Comment (mikkopiu) [inline frontend/src/e2e-test/pages/employee/messaging/messages-page.ts:16]: IMO this should test the select, too.

File: frontend/src/employee-frontend/components/messages/MessageEditor.tsx [hunk 4]
```diff
@@ -17,6 +17,7 @@ import { useTranslation } from '../../state/i18n'
 import { Bulletin } from './types'
 import { SelectorChange } from 'employee-frontend/components/messages/receiver-selection-utils'
 import MultiSelect from 'lib-components/atoms/form/MultiSelect'
+import Select from '../common/Select'
 
 type Option = {
```
Comment (mapehe) [inline frontend/src/employee-frontend/components/messages/MessageEditor.tsx:22]: We might want to refactor this somehow. I don't really like this `Option` guy here. (This is just nit though and it's my code anyway.)

File: frontend/src/employee-frontend/components/messages/MessageEditor.tsx [hunk 5]
```diff
@@ -94,10 +102,19 @@ export default React.memo(function MessageEditor({
...
+          options={mappedSenderOptions}
+          onChange={(selected) =>
+            selected &&
+            'value' in selected &&
+            onChange({ sender: selected.value })
+          }
+          value={
+            mappedSenderOptions.find(
+              ({ value }) => value === bulletin.sender
+            ) || mappedSenderOptions[0]
```
Comment (mapehe) [inline frontend/src/employee-frontend/components/messages/MessageEditor.tsx:115]: Do we want to default to `mappedSenderOptions[0]` here? We could just let it default to `null` and force the user to choose their sender name each time and maybe reduce errors?

**Comment by mapehe** [comment]
LGTM, just a couple of nitty-gritty points.

**Review by mapehe — APPROVED** [review]

**Pull request merged**